pages/api/group/index.ts

```diff
@@ -0,0 +1,113 @@
+import mongoose from 'mongoose';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import {dbConnect} from '@/lib/mongoClient';
+import { responseFormatter } from '@/lib/responseLib';
+import Group from '@/models/Group';
+import  UserGroupRelation  from '@/models/UserGroupRelation';
+
+function getRandomID(name:string):string {
+    let ID = name.split(" ").join("-") + "-";
+    for(let i=0; i<4; i++){
+        ID += Math.floor(Math.random() * 10).toString();
+    }
+    return ID
+  }
+
+export default async function groupHandler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  // Connect with MongoDB database
+  try {
+    await dbConnect();
+  } catch (e) {
+    console.log('fail to connect with database', e);
+    return (
+      res
+        .status(500)
+        .json(responseFormatter(false, null, 'Internal Server Error'))
+    );
+  }
+
+  switch (req.method) {
+    // create a new group
+    case 'POST':
```

Comment:
The `POST` is well written, but there is one issues. Since right now you are conducting multiple cross-collection write operations. One big pitfall is Error happened right after you successfully inserted into the `Group`, but failed to insert the relationship into the `GroupUser`. As a result, client received `500` status code, so it trying to create again, however, that time the group will already be existing, due to the non-atomic-transaction nature for cross-collection operations in MongoDB.

Therefore, to maintain the ACID in MongoDB, you need to use `Transaction`. You could check out these two articles for further studies on Atomicity in MongoDB and how to deploy a transaction in Mongoose. 

[MongoDB Docs](https://www.mongodb.com/docs/manual/core/transactions/)
[Mongoose Docs](https://mongoosejs.com/docs/transactions.html)